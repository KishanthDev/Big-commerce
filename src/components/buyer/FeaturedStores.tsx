export default function FeaturedStores() {
  const stores = [
    {
      name: "Woodland Crafts",
      description:
        "Handcrafted wooden home accessories made from sustainable materials.",
      rating: "4.9 (128 reviews)",
      location: "Portland, OR",
    },
    {
      name: "Modern Threads",
      description:
        "Minimalist clothing with a focus on quality fabrics and ethical production.",
      rating: "4.7 (93 reviews)",
      location: "Austin, TX",
    },
    {
      name: "Tech Innovations",
      description:
        "Unique tech accessories and gadgets for the modern lifestyle.",
      rating: "4.8 (215 reviews)",
      location: "San Francisco, CA",
    },
  ];

  return (
    <section className="dark:bg-black py-16">
      <div className="container mx-auto px-5">
        <h2 className="text-3xl font-semibold dark:text-white text-center mb-12">
          Featured Stores
        </h2>
        <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
          {stores.map((store, index) => (
            <div
              key={index}
              className="bg-white dark:bg-gray-300 rounded-lg overflow-hidden shadow-sm hover:shadow-md hover:-translate-y-1 transition duration-300"
            >
              <div className="bg-gray-200 dark:bg-gray-500 h-48 flex items-center justify-center text-gray-500">
                <span className="dark:text-gray-900">Store Preview</span>
              </div>
              <div className="p-5">
                <h3 className="text-lg font-semibold mb-2">{store.name}</h3>
                <p className="text-gray-600 text-sm mb-4">
                  {store.description}
                </p>
                <div className="flex justify-between text-gray-500 text-xs">
                  <span>‚≠ê {store.rating}</span>
                  <span>üè† {store.location}</span>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
}
