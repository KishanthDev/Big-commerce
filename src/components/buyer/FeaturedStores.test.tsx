import { render, screen } from "@testing-library/react";
import FeaturedStores from "./FeaturedStores";

describe("FeaturedStores", () => {

  test("renders without crashing", () => {
    render(<FeaturedStores />);
    expect(screen.getByText("Featured Stores")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<FeaturedStores />);
    expect(asFragment()).toMatchSnapshot();
  });
});
