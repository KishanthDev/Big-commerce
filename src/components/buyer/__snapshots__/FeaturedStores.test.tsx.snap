// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedStores matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="dark:bg-black py-16"
  >
    <div
      class="container mx-auto px-5"
    >
      <h2
        class="text-3xl font-semibold dark:text-white text-center mb-12"
      >
        Featured Stores
      </h2>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-6"
      >
        <div
          class="bg-white dark:bg-gray-300 rounded-lg overflow-hidden shadow-sm hover:shadow-md hover:-translate-y-1 transition duration-300"
        >
          <div
            class="bg-gray-200 dark:bg-gray-500 h-48 flex items-center justify-center text-gray-500"
          >
            <span
              class="dark:text-gray-900"
            >
              Store Preview
            </span>
          </div>
          <div
            class="p-5"
          >
            <h3
              class="text-lg font-semibold mb-2"
            >
              Woodland Crafts
            </h3>
            <p
              class="text-gray-600 text-sm mb-4"
            >
              Handcrafted wooden home accessories made from sustainable materials.
            </p>
            <div
              class="flex justify-between text-gray-500 text-xs"
            >
              <span>
                ‚≠ê 4.9 (128 reviews)
              </span>
              <span>
                üè† Portland, OR
              </span>
            </div>
          </div>
        </div>
        <div
          class="bg-white dark:bg-gray-300 rounded-lg overflow-hidden shadow-sm hover:shadow-md hover:-translate-y-1 transition duration-300"
        >
          <div
            class="bg-gray-200 dark:bg-gray-500 h-48 flex items-center justify-center text-gray-500"
          >
            <span
              class="dark:text-gray-900"
            >
              Store Preview
            </span>
          </div>
          <div
            class="p-5"
          >
            <h3
              class="text-lg font-semibold mb-2"
            >
              Modern Threads
            </h3>
            <p
              class="text-gray-600 text-sm mb-4"
            >
              Minimalist clothing with a focus on quality fabrics and ethical production.
            </p>
            <div
              class="flex justify-between text-gray-500 text-xs"
            >
              <span>
                ‚≠ê 4.7 (93 reviews)
              </span>
              <span>
                üè† Austin, TX
              </span>
            </div>
          </div>
        </div>
        <div
          class="bg-white dark:bg-gray-300 rounded-lg overflow-hidden shadow-sm hover:shadow-md hover:-translate-y-1 transition duration-300"
        >
          <div
            class="bg-gray-200 dark:bg-gray-500 h-48 flex items-center justify-center text-gray-500"
          >
            <span
              class="dark:text-gray-900"
            >
              Store Preview
            </span>
          </div>
          <div
            class="p-5"
          >
            <h3
              class="text-lg font-semibold mb-2"
            >
              Tech Innovations
            </h3>
            <p
              class="text-gray-600 text-sm mb-4"
            >
              Unique tech accessories and gadgets for the modern lifestyle.
            </p>
            <div
              class="flex justify-between text-gray-500 text-xs"
            >
              <span>
                ‚≠ê 4.8 (215 reviews)
              </span>
              <span>
                üè† San Francisco, CA
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
